ibl_pipeline/analyses/behavior.py

```diff
@@ -6,6 +6,69 @@
 schema = dj.schema(dj.config.get('database.prefix', '') + 'ibl_analyses_behavior')
 
 
+@schema
+class Contrasts(dj.Computed):
+    definition = """
+    -> behavior.TrialSet
+    ---
+    contrasts:                      blob    # contrasts used in this session, negative when on the left
+    has_no_contrast:                bool    # have trials with no contrast
+    has_left:                       bool    # have trials with contrast on the left
+    has_right:                      bool    # have trials with contrast on the right
+    contrasts_left=null:            blob    # contrasts shown on the left
+    contrasts_right=null:           blob    # contrasts shown on the right
+    n_trials_stim:                  blob    # number of trials for each contrast
+    n_trials_stim_left=null:        blob    # number of trials when the stimulus is shown on the left
+    n_trials_stim_right=null:       blob    # number of trials when the stimulus is shown on the right
+    n_trials_stim_no_contrast=null: int     # number of trials when no stimulus is shown 
+    """
+
+    def make(self, key):
+        print(key)
+        key_con = key.copy()
+        trials = behavior.TrialSet.Trial & key
+        contrasts_left = np.unique(trials.fetch('trial_stim_contrast_left'))[1:] # discard contrast 0
+        contrasts_right = np.unique(trials.fetch('trial_stim_contrast_right'))[1:] # discard contrast 0
+        trials_no_contrast = trials & 'trial_stim_contrast_right=0' & 'trial_stim_contrast_left=0'
+
+        key_con['has_left'] = False
+        key_con['has_right'] = False
+        key_con['has_no_contrast'] = False
+        
+        if contrasts_left.size != 0:
+            key_con['has_left'] = True
+            key_con['contrasts_left'] = contrasts_left
+            n_trials_stim_left = [len(trials & 'trial_stim_contrast_left={}'.format(contrast)) \
+                                    for contrast in contrasts_left]
+            key_con['n_trials_stim_left'] = n_trials_stim_left
+        else:
+            n_trials_stim_left = []
+            
+        if contrasts_right.size != 0:
+            key_con['has_right'] = True
+            key_con['contrasts_right'] = contrasts_right
+            n_trials_stim_right = [len(trials & 'trial_stim_contrast_right={}'.format(contrast)) \
```

Comment:
Why use string conditions instead of dict conditions? `trials & {'trial_stim_contrast_right': contrast}`

Reply:
Also, this loop can be replaced with a single `dj.U` query:
```
n_trials_stim_right = dj.U('contrast_right').aggr(trials, n='count(*)').fetch('n')
```